runtime/src/config.c

```diff
@@ -448,7 +449,7 @@ void parseConfigFile(const char* configFilename,
     char setConfigBuffer[_default_string_length];
     numScans = fscanf(argFile, _default_format_read_string, setConfigBuffer);
     if (numScans == 1) {
-      if (!aParsedString(argFile, setConfigBuffer, 0, configFilename)) {
+      if (!aParsedString(argFile, setConfigBuffer, 0, filename)) {
```

Comment:
Darn, this is unfortunate.  We really need to print messages referring to the configFilename here, because we're talking about the user's config var settings in the file they named on the command line.  Mentioning internal runtime files or worse yet, no file at all, won't be helpful.  How about expanding the error.c interfaces so that in addition to chpl_error() and the like that take filename indices, we also have (say) chpl_error_fn_explicit() that takes a filename as const char*?  Then call those for the cases here where we want to mention configFilename.  (In fact, the rewritten chpl_error() and the like could be a fairly thin interface over the new _fn_explicit() functions, or whatever you call them.)